sky/packages/sky/lib/sky_tool

```diff
@@ -254,6 +284,359 @@ class StopSky(object):
         pids.clear()
 
 
+class IOSDevice(object):
+    _has_ios_deploy = None
+    @classmethod
+    def has_ios_deploy(cls):
+        if cls._has_ios_deploy is not None:
+            return cls._has_ios_deploy
+        cmd = [
+            'which',
+            'ios-deploy'
+        ]
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+        out, err = process.communicate()
+        match = re.search(r'ios-deploy', out)
+        cls._has_ios_deploy = match is not None
+        return cls._has_ios_deploy
+
+    _is_connected = False
+    @classmethod
+    def is_connected(cls):
+        if not cls.has_ios_deploy():
+            return False
+        if cls._is_connected:
```

Comment:
_is_connected is only supposed to hold a boolean result, which is why it's initialized to False above.